main.go

```diff
@@ -215,3 +221,30 @@ func ensureManagedOCS(c client.Client, log logr.Logger, envVars map[string]strin
 		return err
 	}
 }
+
+func checkIfCRDExist(crdNames []string) map[string]bool {
+
+	var options client.Options
+	options.Scheme = runtime.NewScheme()
+	utilruntime.Must(apiextensionsv1.AddToScheme(options.Scheme))
+
+	k8sClient, err := client.New(config.GetConfigOrDie(), options)
+	if err != nil {
+		setupLog.Error(err, "error creating client")
+		os.Exit(1)
+	}
+
+	CRDExist := map[string]bool{}
+	for _, crdName := range crdNames {
+		crd := apiextensionsv1.CustomResourceDefinition{}
+		err = k8sClient.Get(context.Background(), types.NamespacedName{Name: crdName}, &crd)
+
+		if err != nil && !errors.IsNotFound(err) {
+			setupLog.Error(err, fmt.Sprintf("Error Getting CRD for %s", crdName))
+			os.Exit(1)
+		}
+
+		CRDExist[crdName] = crd.UID != ""
```

Comment:
- is there any scenario where both the CRDs exist?
- I remembering hearing that a previous running cluster with SDN will continue to use SDN even though OVN exists
- however, this needs to be confirmed.

Reply:
No there isn't! but this is a util function taking any number of CRDs and can be useful for other feature that need CRD detection.